bitcoin-rpc-provider/src/lib.rs

```diff
@@ -168,7 +172,61 @@ fn enc_err_to_manager_err(_e: EncodeError) -> ManagerError {
     Error::BitcoinError.into()
 }
 
-impl Signer for BitcoinCoreProvider {
+impl ContractSignerProvider for BitcoinCoreProvider {
+    type Signer = SimpleSigner;
+
+    fn derive_signer_key_id(&self, _is_offer_party: bool, temp_id: [u8; 32]) -> [u8; 32] {
+        temp_id // fixme not safe
+    }
+
+    fn derive_contract_signer(&self, keys_id: [u8; 32]) -> Result<Self::Signer, ManagerError> {
+        let label_map = self
+            .client
+            .lock()
+            .unwrap()
+            .call::<HashMap<Address, Value>>(
+                "getaddressesbylabel",
+                &[Value::String(keys_id.to_hex())],
+            )
+            .map_err(rpc_err_to_manager_err)?;
+
+        // we should only have one address per keys_id
+        // if not something has gone wrong
+        assert_eq!(label_map.len(), 1);
```

Comment:
This will throw when the private key is not yet created. Either create it in the `derive_signer_key_id` or assert that it's less than or equal to 1. It's unfortunate that this crate is not tested...

Reply:
https://github.com/p2pderivatives/rust-dlc/issues/200

Reply:
Woops, good catch. Is this not used in the manager tests?

Reply:
No it's not since I've changed to use `SimpleWallet` and electrs. I'm considering getting rid of it as it's a maintenance burden. But thanks for fixing this!